packages/vite/src/utils/helper-functions.ts

```diff
@@ -0,0 +1,43 @@
+import { defineConfig, UserConfig, UserConfigExport } from 'vite';
+
+export function getViteFrameworkPluginPath(uiFramework: string) {
+  const frameworkPaths = {
+    react: '@vitejs/plugin-react',
+    vue3: '@vitejs/plugin-vue',
+    'vue3-jsx': '@vitejs/plugin-vue-jsx',
+  };
+  return frameworkPaths[uiFramework];
+}
+
+export async function buildModeConfigSetup(
+  config: UserConfigExport,
+  mode: string
+): Promise<UserConfig> {
+  if (typeof config === 'function') {
+    return config({ command: 'build', mode });
+  }
+  return config;
+}
+
+export async function baseConfigWithPlugins(
+  uiFramework: string,
+  projectRoot: string
+): Promise<UserConfigExport> {
+  const frameworkPlugin = (
+    await import(getViteFrameworkPluginPath(uiFramework))
+  ).default;
+  const viteTsConfigPathsPluginName = 'vite-tsconfig-paths';
+  const viteTsConfigPathsPlugin = (await import(viteTsConfigPathsPluginName))
+    .default;
+  return defineConfig(() => {
+    return {
+      plugins: [
+        frameworkPlugin(),
+        viteTsConfigPathsPlugin({
+          root: projectRoot,
+          projects: ['tsconfig.app.json'],
```

Comment:
We should point to the root `tsconfig.base.json` since that is the only place we add import paths for workspace libs to. The `tsconfig.app.json` _will_ work because it extends from root, but let's just point to root.

Reply:
🤦🏻‍♀️🤦🏻‍♀️ You just saved me 1 hour of coding :P `tsconfig.app.json` does NOT work for deep imports, and I was thinking the best way to do this. Somehow, adding `tsconfig.base.json` there did NOT occur to me. I was on a path of "parse tsconfig.base.json and then add the paths there manualy". 

So, yes, can confirm, `tsconfig.base.json` works.